### DISCUSSION TITLE
a socket write failed, the socket might already be closed

### DISCUSSION THREAD

**Discussion by shifeichen** [rabbitmq/rabbitmq-server#5261] [Other]
I installed rabbitmq on visual machines, and I found some errors in rabbitmq log file,  It  is printing continuously  "a socket write failed, the socket might already be closed". 

Can you give me some advice for arranging mistakes??

**Comment by shifeichen** [comment]
2022-07-18 09:15:54.314834+08:00 [error] <0.8505.0> MQTT: a socket write failed, the socket might already be closed

**Reply by michaelklishin** [reply]
Most likely your MQTT clients abruptly close TCP connections very quickly so RabbitMQ doesn't have a chance to send the final frame such as the Last Will one. On closing connections, this is basically log noise.
